components/SelectedRestaurants.js

```diff
@@ -1,31 +1,57 @@
  import RestaurantCard from './RestaurantCard.js';
  import foodSrc from '../public/images/food-img.jpg';
+ import Router from 'next/router';
+
 
 const arrOfRestaurants = [];
 
 for (let i = 0; i < 5; i++) {
   const obj = {
     id: i,
     name: "Wing Lum Cafe",
+    restaurantName: "Wing Lum Cafe",
     category: "Chinese",
     description: 'description',
     deliverySpeed: "fast",
     avgCost: "$$$",
     distance:"12",
     imgSrc: foodSrc,
-    rating: "3.4"
+    rating: "3.4",
+    dishes: [
+      {
+          photoURL: "https://s3-media0.fl.yelpcdn.com/bphoto/-vpD8rR9-v61LpTG9tXiWg/o.jpg",
+          name: "Amatriciana"
+      },
+      {
+          name: "Moto",
+          photoURL: "https://s3-media0.fl.yelpcdn.com/bphoto/9_ZCRhfIUAsj-oeXFFidkQ/o.jpg"
+      },
+      {
+          photoURL: "https://s3-media0.fl.yelpcdn.com/bphoto/yvvLJoKtaSr7z4IUvViPKg/o.jpg",
+          name: "Bianca"
+      }
+  ],
   };
   arrOfRestaurants.push(obj);
 };
 
+
+
+function restaurantClickHandler(restaurant){
+  console.log('restaurant clickHandler with restaurant', restaurant);
+  Router.push({
+    pathname: '/Page-J-Restaurant',
+    query: { name: restaurant.restaurantName}
+  }, '/Page-J-Restaurant');
+}
+
 const topThree = arrOfRestaurants.map((restaurant, index) => {
   return (
-    <div>
-      <RestaurantCard key={restaurant.id} restaurant={restaurant}/>
-    </div>
+    <>
+      <RestaurantCard key={index} restaurant={restaurant} ClickHandler={()=>restaurantClickHandler(restaurant)}/>
+    </>
   )
 });
-
 export default function Favorite() {
```

Comment:
Component name should match the filename or we get really confusing stack traces when there's an error.

Reply:
done